apps/rewards/app/components/App/App.js

```diff
@@ -41,6 +48,27 @@ class App extends React.Component {
     }
   }
 
+  componentWillReceiveProps(nextProps) {
+    if(nextProps.balances) {
+      this.updateConvertedRates(nextProps)
```

Comment:
Since this is an asynchronous call I think it should be handled in a different component lifecycle call to prevent multiple renders (componentDidUpdate) although I guess if the goal is to re-trigger conversion updates that's an interesting approach? Might want to set a timer in the componentdidmount lifecycle and just retrigger conversions like every 30 seconds or something idk. I guess it depends how accurate we want the conversions to be.